ibis/expr/api.py

```diff
@@ -2472,6 +2473,8 @@ def _table_drop(self, fields):
     join=join,
     cross_join=cross_join,
     inner_join=_regular_join_method('inner_join', 'inner'),
+    inner_semi_join=_regular_join_method('inner_semi_join', 'inner_semi'),
+    left_semi_join=_regular_join_method('left_semi_join', 'left_semi'),
```

Comment:
Per my below comments, remove these and bring back the `any_left/any_inner` (`all_*` isn't necessary because those have the conventional join semantics.)